### PR TITLE
Allow schema loader to output loading statement to standard output

### LINKED ISSUE #122: Allow schema loader to output loading statement to standard output
Rather than directly connecting to a database it is useful to request outputting SQL statements to standard output. This may be done by accepting - (a dash) as the <database name> parameter.

Having this support would help outermost loaders to load multiple schema supports in a single transaction, which would otherwise be impossible with the loader.

See also https://github.com/linz/linz-lds-bde-schema/issues/137

### PR DESCRIPTION
**Pull request by strk**
Also ensures transactional operations
Closes #122

Print progress messages to stderr instead of stdout
Includes ChangeLog entry

### CODE REVIEW

**Comment by strk** [comment]
Yeah, that depends on the fact that the installed dbpatch does not support stdout loading, and the changes in this PR assume it does. It will take some effort to forbid stdout-based approach unless all underlying tools support it

**Comment by strk** [comment]
With latest commit I restored support for pre-stdout supporting table_version and dbpatch loaders. Now Travis is happy, but of course the loader is NOT transactional. A WARNING message is printed in those cases.

**Comment by strk** [comment]
Alright we now also have tests. With stdout loader only tested with PosgreSQL 9.6 and PostGIS 2.3 but I don't see why it shouldn't work for the other matrix entries anyway

File: Makefile [hunk 1]
```diff
@@ -92,6 +94,47 @@ installcheck:
...
 
+# TODO: run the full test after preparing the db ?
+check-loader-stdout:
+
+	dropdb --if-exists linz-bde-schema-test-db
+
+	createdb linz-bde-schema-test-db
+	linz-bde-schema-load - | \
+        psql -Xo /dev/null linz-bde-schema-test-db
+	linz-bde-schema-load - | \
```
Comment (palmerj) [inline Makefile]: Duplicated line?

Comment (strk) [reply Makefile]: It is actually intentional, we test that the script is idempotent, basically.
I'd like to reduce the overall duplication though, one day...

File: Makefile [hunk 2]
```diff
@@ -92,6 +94,47 @@ installcheck:
...
+        psql -Xo /dev/null linz-bde-schema-test-db
+	export PGDATABASE=linz-bde-schema-test-db; \
+	V=`psql -XtAc 'select bde.bde_version()'` && \
+	echo $$V && test "$$V" = "$(VERSION)" && \
+	V=`linz-bde-schema-load --version` && \
+	echo $$V && test `echo "$$V" | awk '{print $$1}'` = "$(VERSION)"
+	dropdb linz-bde-schema-test-db
+
+	createdb linz-bde-schema-test-db
+	linz-bde-schema-load --noextension - | \
```
Comment (palmerj) [inline Makefile]: As above

File: Makefile [hunk 3]
```diff
@@ -92,6 +94,47 @@ installcheck:
...
+        psql -Xo /dev/null linz-bde-schema-test-db
+	export PGDATABASE=linz-bde-schema-test-db; \
+	V=`psql -XtAc 'select bde.bde_version()'` && \
+	echo $$V && test "$$V" = "$(VERSION)" && \
+	V=`linz-bde-schema-load --version` && \
+	echo $$V && test `echo "$$V" | awk '{print $$1}'` = "$(VERSION)"
+	dropdb linz-bde-schema-test-db
+
+	createdb linz-bde-schema-test-db
+	linz-bde-schema-load --revision - | \
```
Comment (palmerj) [inline Makefile]: As above

**Pull request merged**